### PR TITLE
Unify rendering into helper

### PR DESCRIPTION
**Pull request by jyruzicka**
Split `Tracks::Utils` into two parts. Filename sanitization is extracted into `DataController`, while rendering is shifted into a helper. Add tests for rendering, remove tests that rely on `rendered_notes`. In response to Issue #1773.

### CODE REVIEW

**Comment by lrbalt** [comment]
look great, but could you 
* remove the update to gemfile. That should go in a separate commit
* create a migration to remove the rendered_note from the model (and add it for the rollback)

**Comment by dnrce** [comment]
I fixed Gemfile.lock in master. That change can be removed from this PR.

**Comment by dnrce** [comment]
Please clean up your code. To start with, the indentation is very inconsistent.

**Comment by jyruzicka** [comment]
@dnrce Can fix indentation. Any other sections which need cleanup? Additionally, best to submit further changes as new commits, and rebase once everything is in order?

**Comment by dnrce** [comment]
Rebasing now is fine. I think in this case it will make it easier to review.

**Comment by jyruzicka** [comment]
To my surprise, I appear to have pushed changes and I don't think anything has broken.

**Comment by lrbalt** [comment]
@dnrce Dan, I think it is save to merge. Do you want this in 2.3_branch too? I think we should only merge in master, but I'd like your opinion

**Comment by dnrce** [comment]
This should *not* go into 2.3 prior to release.

**Comment by mattr-** [comment]
This needs another rebase.

**Comment by mattr-** [comment]
Closing this for now. I like the approach, but I'm not sure what's going on in the down migration path and whether or not it's needed (I'm sure I'm just missing something), it needs a rebase, and has sat untouched for almost 5 months.

**Comment by jyruzicka** [comment]
Hey @mattr- - last few months have been a bit busy, and it's been ages since I've looked at this, but I have a little free time and if you think this is worth including in 2.4.0 I can give it a look over, work out what's going on with that down migration, and fix up the rebase. Worth opening up again?

**Comment by mattr-** [comment]
Yup, I'd like to get this included, if possible.

File: db/migrate/20150413194512_remove_rendered_notes.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def self.up
+    remove_column :todos, 'rendered_notes'
+  end
+
+  def self.down
+    add_column :todos, 'rendered_notes', :text
+    
+    Todo.reset_column_information
+
+    say "Clearing show_from dates from completed todos"
```
Comment (mattr-) [inline db/migrate/20150413194512_remove_rendered_notes.rb]: The thing that confuses me about this migration is that we're clearing the `show_from` dates for completed todos, which doesn't seem strictly necessary. Is this something that just needs to be done for data cleanliness and doesn't really need to be in this PR?

**Comment by jyruzicka** [comment]
Had a look over this last night: I think that migration was just complete fail on my behalf. I'd copied the code for rendering todo notes from the up migration of `20120412072508_add_rendered_notes` and completely blanked on the fact that there was a bunch of other code in that migration. I've tidied that up and rebased against master locally. I have one test failing my end, so once I work out why I'll push.

**Comment by jyruzicka** [comment]
Just pushed the new copy.

**Comment by mattr-** [comment]
I like it. Thanks @jyruzicka!

File: db/migrate/20150413194512_remove_rendered_notes.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    remove_column :todos, 'rendered_notes'
+  end
+
+  def self.down
+    add_column :todos, 'rendered_notes', :text
+    
+    # Call save on each todo to force generation of rendered_notes
+    # Copied from 20120412072508_add_rendered_notes
+    say "Rendering todo notes..."
+    Todo.all.each{ |todo| todo.save(validate: false) }
```
Comment (dnrce) [inline db/migrate/20150413194512_remove_rendered_notes.rb]: I'd rather consider this an irreversible migration than depend on the model in a migration.

**Pull request merged**